File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        int numWaysToSplitArray = 0;
        // Calculating the sum of the `nums` array
        long sum = 0;
        for(int num : nums){
            sum += num;
        }

        long cummulativeLeftSum = 0;
        for(int i = 0; i < nums.length - 1; i++){
            cummulativeLeftSum += nums[i];

            if(cummulativeLeftSum >= sum - cummulativeLeftSum) numWaysToSplitArray++;
        }
        System.out.println(sum);
        
        return numWaysToSplitArray;
    }
}

File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        int count = 0;
        // Precalculating the sum of the array
        long long sum = 0;
        for(const auto &num : nums){
            sum += num;
        }

        long long accumulator = 0;
        for(int i = 0; i < nums.size() - 1; i++){
            accumulator += nums[i];

            if(accumulator >= sum - accumulator){
                count++;
            }
        }

        return count;
    }
};